frontend/src/metabase/actions/containers/ActionCreator/FormCreator/FormCreator.unit.spec.tsx

```diff
@@ -237,4 +242,97 @@ describe("actions > containers > ActionCreator > FormCreator", () => {
       defaultValue,
     );
   });
+
+  describe("Warning banner", () => {
+    it("shows a warning banner", () => {
+      const defaultValue = "foo bar";
+      const parameter = makeParameter({ required: true });
+      const fieldSettings = makeFieldSettings({
+        inputType: "string",
+        required: true,
+        defaultValue,
+        hidden: true,
+      });
+
+      setup({
+        parameters: [parameter],
+        formSettings: {
+          type: "form",
+          fields: {
+            [parameter.id]: fieldSettings,
+          },
+        },
+        isPublic: true,
+      });
+
+      expect(screen.getByTestId("action-warning-banner")).toBeInTheDocument();
+    });
+
+    it("shows a banner for implicit action params", () => {
+      const parameter = makeParameter({ required: true });
+      const fieldSettings = createMockImplicitActionFieldSettings({
+        id: parameter.id,
+        hidden: true,
+      });
+
+      setup({
+        parameters: [parameter],
+        formSettings: {
+          type: "form",
+          fields: {
+            [parameter.id]: fieldSettings,
+          },
+        },
+        isPublic: true,
+      });
+
+      expect(screen.getByTestId("action-warning-banner")).toBeInTheDocument();
+    });
+
+    it("Doesn't show a warning banner for not public action", () => {
```

Comment:
nit: matches character case to other tests and fixes typo
```suggestion
    it("doesn't show a warning banner for nonpublic action", () => {
```